### ISSUE TITLE
New lines in Python code causes different behavior

### ISSUE THREAD

**Issue by ddkang** [pytorch/pytorch#15053]
## 🐛 Bug

In the forward function, adding new lines completely changes behavior (50% vs 85% accuracy on a binary classification task).

## To Reproduce

Change the torchvision resnet's basic block to this:
```
    def forward(self, x): 
        residual = x 
        out = self.conv1(x)
        out = F.dropout(out, training=True)
        out = self.relu(out)
        out = self.conv2(out)
        out = F.dropout(out, training=True)
        if self.downsample is not None:
            residual = self.downsample(x)
        out += residual
        out = self.relu(out)
        return out 
```

This works as expected (the _only_ difference are the new lines):
```
    def forward(self, x): 
        residual = x 

        out = self.conv1(x)
        out = F.dropout(out, training=True)
        out = self.relu(out)

        out = self.conv2(out)
        out = F.dropout(out, training=True)

        if self.downsample is not None:
            residual = self.downsample(x)

        out += residual
        out = self.relu(out)
        return out 
```

## Expected behavior

New lines should not affect Module behavior.

## Environment

Collecting environment information...
PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.4 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.10.0

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: TITAN V
GPU 1: TITAN V
GPU 2: TITAN V
GPU 3: TITAN V

Nvidia driver version: 390.42
cuDNN version: Probably one of the following:
/usr/local/cuda-9.0/lib64/libcudnn.so.7.0.4
/usr/local/cuda-9.0/lib64/libcudnn_static.a

Versions of relevant libraries:
[pip3] intel-numpy (1.15.1)
[pip3] numpy (1.15.1)
[pip3] torch (0.4.1)
[pip3] torchfile (0.1.0)
[pip3] torchvision (0.2.1)
[conda] Could not collect

**Comment by albanD** [comment]
Hi,

Could you give a full sample to reproduce please? These two codes give exactly the same result for me.
